torch/utils/data/datapipes/iter/combinatorics.py

```diff
@@ -39,7 +39,7 @@ def __init__(self,
         self.sampler_args = () if sampler_args is None else sampler_args
         self.sampler_kwargs = {} if sampler_kwargs is None else sampler_kwargs
         # https://github.com/python/mypy/pull/9629 will solve
-        self.sampler = sampler(data_source=self.datapipe, *self.sampler_args, **self.sampler_kwargs)  # type: ignore[misc]
+        self.sampler = sampler(*self.sampler_args, **self.sampler_kwargs, data_source=self.datapipe)  # type: ignore[misc]
```

Comment:
Do we expect the user to be able to override the data_source with kwargs?

Reply:
@Skylion007 I checked the rules and seems like we can put kwargs after the `data_source`

```
- self.sampler = sampler(data_source=self.datapipe, *self.sampler_args, **self.sampler_kwargs)  # type: ignore[misc]
+ self.sampler = sampler(*self.sampler_args, data_source=self.datapipe, **self.sampler_kwargs)  # type: ignore[misc]
```
So I will change this for all the file changes. Also should we check args for the overrides?

Reply:
I'f feel a bit more comfortable with this as it's less likely to be a behavior change.

Reply:
Sure so lets add an assert for the `data_source` and other arguments in kwargs

Reply:
Sure so let me make the changes